toolbox/include/CentroidalMomentum.h

```diff
@@ -1,35 +1,39 @@
 #ifndef WBT_CENTROIDALMOMENTUM_H
 #define WBT_CENTROIDALMOMENTUM_H
 
-#include "WBIModelBlock.h"
+#include "WBBlock.h"
+#include <memory>
 
 namespace wbt {
     class CentroidalMomentum;
 }
 
-class wbt::CentroidalMomentum : public wbt::WBIModelBlock {
+namespace iDynTree {
+    class SpatialMomentum;
+}
 
-    double *m_basePose;
-    double *m_centroidalMomentum;
+class wbt::CentroidalMomentum : public wbt::WBBlock
+{
+private:
+    std::unique_ptr<iDynTree::SpatialMomentum> m_centroidalMomentum;
```

Comment:
There is probably a reason, but I don't remember it: why it is not possible to declare this variable simply as class members?

Reply:
Headers cleanliness I'd say. This was the pattern used in the previous release, and I kept it also in this new version.